cocotb/decorators.py

```diff
@@ -87,38 +94,38 @@ class RunningCoroutine(object):
         coro.kill() will destroy a coroutine instance (and cause any Join
         triggers to fire.
     """
-    def __init__(self, inst, parent):
-        if hasattr(inst, "__name__"):
-            self.__name__ = "%s" % inst.__name__
+
+    def __init__(self, inst, parent=None):
 
         if sys.version_info[:2] >= (3, 5) and inspect.iscoroutine(inst):
             self._natively_awaitable = True
             self._coro = inst.__await__()
-        else:
+            self._funcname = inst.__name__
+            self._module = modulename_from_filename(inst.cr_code.co_filename)
+            self.__doc__ = inst.__doc__
```

Comment:
To be clear here: I don't think we need to keep these properties when parent=None, only for existing users where parent is not none (and where extracting them is trivial)

Reply:
And what will the behavior for `funcname` and `module` properties be? `AttributeError`? I'm not against it, just figured it would be considered a less optimal solution.

Reply:
Put an alternative approach in #1259, let me know what you think, and I'll close that PR if you argue against it or integrate the approach into this PR.